pkg/spec/controller_test.go

```diff
@@ -0,0 +1,234 @@
+  - port: 8082`),
+			App: &DeploymentSpecMod{
+				ControllerFields: ControllerFields{
+
+					Name: "test",
+					PodSpecMod: PodSpecMod{
+						Containers: []Container{{Container: api_v1.Container{Name: "test", Image: "nginx"}}},
+					},
+					Services: []ServiceSpecMod{
+						{
+							Name: "nginx",
+							Ports: []ServicePortMod{
+								{ServicePort: api_v1.ServicePort{Port: 8080, Name: "nginx-8080"}},
+								{ServicePort: api_v1.ServicePort{Port: 8081, Name: "nginx-8081"}},
+								{ServicePort: api_v1.ServicePort{Port: 8082, Name: "nginx-8082"}},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			Name: "Multiple ports, some with names specified, others with no names",
+			Data: []byte(`
+name: test
+containers:
+ - image: nginx
+services:
+- ports:
+  - port: 8080
+    name: port-1
+  - port: 8081
+    name: port-2
+  - port: 8082
+  - port: 8083`),
+			App: &DeploymentSpecMod{
+				ControllerFields: ControllerFields{
+
+					Name: "test",
+					PodSpecMod: PodSpecMod{
+						Containers: []Container{{Container: api_v1.Container{Name: "test", Image: "nginx"}}},
+					},
+					Services: []ServiceSpecMod{
+						{
+							Name: "test",
+							Ports: []ServicePortMod{
+								{ServicePort: api_v1.ServicePort{Port: 8080, Name: "port-1"}},
+								{ServicePort: api_v1.ServicePort{Port: 8081, Name: "port-2"}},
+								{ServicePort: api_v1.ServicePort{Port: 8082, Name: "test-8082"}},
+								{ServicePort: api_v1.ServicePort{Port: 8083, Name: "test-8083"}},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			Name: "Multiple ports, all with names",
+			Data: []byte(`
+name: test
+containers:
+ - image: nginx
+services:
+- ports:
+  - port: 8080
+    name: port-1
+  - port: 8081
+    name: port-2
+  - port: 8082
+    name: port-3`),
+			App: &DeploymentSpecMod{
+				ControllerFields: ControllerFields{
+
+					Name: "test",
+					PodSpecMod: PodSpecMod{
+						Containers: []Container{{Container: api_v1.Container{Name: "test", Image: "nginx"}}},
+					},
+					Services: []ServiceSpecMod{
+						{
+							Name: "test",
+							Ports: []ServicePortMod{
+								{ServicePort: api_v1.ServicePort{Port: 8080, Name: "port-1"}},
+								{ServicePort: api_v1.ServicePort{Port: 8081, Name: "port-2"}},
+								{ServicePort: api_v1.ServicePort{Port: 8082, Name: "port-3"}},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			Name: "Single port, without any name",
+			Data: []byte(`
+name: test
+containers:
+ - image: nginx
+services:
+- ports:
+  - port: 8080`),
+			App: &DeploymentSpecMod{
+				ControllerFields: ControllerFields{
+
+					Name: "test",
+					PodSpecMod: PodSpecMod{
+						Containers: []Container{{Container: api_v1.Container{Name: "test", Image: "nginx"}}},
+					},
+					Services: []ServiceSpecMod{
+						{Name: "test", Ports: []ServicePortMod{{ServicePort: api_v1.ServicePort{Port: 8080}}}},
+					},
+				},
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.Name, func(t *testing.T) {
```

Comment:
Test code of this function and `ControllerFields`'s code looks similar. We can just call `CoreOperations`? Rather than writing it's code here?

Reply:
CoreOperations also calls the Transform method, while that is not being tested here.
Since this is a refactor, I've not added new tests for Transform method.
We will need to add tests for CoreOperations in subsequent unrelated PRs.

Reply:
ahh ok, now I see it!